java-checks/src/main/java/org/sonar/java/checks/MembersDifferOnlyByCapitalizationCheck.java

```diff
@@ -203,4 +228,28 @@ private static boolean isVariableToExtract(Symbol symbol) {
   private static boolean isMethodToExtract(Symbol symbol) {
     return symbol.isMethodSymbol() && !"<init>".equals(symbol.name());
   }
+
+  private static class ReturnVisitor extends BaseTreeVisitor {
+
+    private final Symbol variableSymbol;
+    private boolean returnsVariable;
+    private int returnCount;
+
+    ReturnVisitor(Symbol variableSymbol) {
+      this.variableSymbol = variableSymbol;
+    }
+
+    @Override
+    public void visitReturnStatement(ReturnStatementTree tree) {
+      returnCount++;
+      ExpressionTree returnExpression = tree.expression();
+      if (returnExpression != null && returnExpression.is(Tree.Kind.IDENTIFIER)) {
+        returnsVariable = ((IdentifierTree) returnExpression).symbol().equals(variableSymbol);
+      }
+    }
+
+    boolean singleReturnWithVariableSymbol() {
+      return returnCount == 1 && returnsVariable;
```

Comment:
Yes, there can be multiple statement as far as there is only one return statement returning the field. I added the test case, thanks for spotting that. However, I think the methods should not have more than 1 return statement, and I find using the visitor as easiest way to assure that.